### ISSUE TITLE
Tests don't pass consistently

### ISSUE THREAD

**Issue by tarcieri** [httprb/http.rb#230]
Would be nice if we had a strategy here, perhaps:

1. Tests that don't pass consistently get flagged as such (i.e. pending), perhaps on a VM by VM basis (i.e. JRuby)
2. Perhaps use e.g. rspec-retry?

It would be nice if there were no transient test failures

**Comment by zanker** [comment]
What are the failures? I think it depends on what they are mostly since that can affect whether it's easy to fix or not.

**Comment by tarcieri** [comment]
https://travis-ci.org/httprb/http.rb/builds

They're all on JRuby, e.g.:

https://travis-ci.org/httprb/http.rb/jobs/62103206

```
  1) HTTP::Client working with SSL timeouts with a per operation timeout read of 0 times out
     Failure/Error: expect { response }.to raise_error(HTTP::TimeoutError, /Read/i)
       expected HTTP::TimeoutError with message matching /Read/i but nothing was raised
     Shared Example Group: "HTTP handling" called from ./spec/lib/http/client_spec.rb:219
     # /home/travis/.rvm/gems/jruby-1.7.19-d19/gems/rspec-expectations-3.2.1/lib/rspec/expectations/fail_with.rb:29:in `fail_with'
     # /home/travis/.rvm/gems/jruby-1.7.19-d19/gems/rspec-expectations-3.2.1/lib/rspec/expectations/handler.rb:40:in `handle_failure'
     # /home/travis/.rvm/gems/jruby-1.7.19-d19/gems/rspec-expectations-3.2.1/lib/rspec/expectations/handler.rb:50:in `handle_matcher'
     # /home/travis/.rvm/gems/jruby-1.7.19-d19/gems/rspec-expectations-3.2.1/lib/rspec/expectations/handler.rb:27:in `with_matcher'
     # /home/travis/.rvm/gems/jruby-1.7.19-d19/gems/rspec-expectations-3.2.1/lib/rspec/expectations/handler.rb:48:in `handle_matcher'
     # /home/travis/.rvm/gems/jruby-1.7.19-d19/gems/rspec-expectations-3.2.1/lib/rspec/expectations/expectation_target.rb:54:in `to'
     # /home/travis/.rvm/gems/jruby-1.7.19-d19/gems/rspec-expectations-3.2.1/lib/rspec/expectations/expectation_target.rb:87:in `to'
     # ./spec/support/http_handling_shared.rb:53:in `(root)'
```

https://travis-ci.org/httprb/http.rb/jobs/63726652

```
  1) HTTP::Client working with SSL connection reuse when disabled opens new sockets
     Failure/Error: client.get("#{server.endpoint}/socket/2").body.to_s
     IOError:
       problem making HTTP request: End of file reached
     Shared Example Group: "HTTP handling" called from ./spec/lib/http/client_spec.rb:219
     # ./lib/http/connection.rb:99:in `read_headers!'
     # ./lib/http/client.rb:77:in `make_request'

  2) HTTP::Client working with SSL connection reuse when enabled on a mixed state re-opens the connection
     Failure/Error: second_socket_id = client.get("#{server.endpoint}/socket/2").body.to_s
     IOError:
       problem making HTTP request: End of file reached
     Shared Example Group: "HTTP handling" called from ./spec/lib/http/client_spec.rb:219
     # ./lib/http/connection.rb:99:in `read_headers!'
     # ./lib/http/client.rb:77:in `make_request'
```

https://travis-ci.org/httprb/http.rb/jobs/63730552

```
  1) HTTP::Client working with SSL connection reuse when enabled with a socket issue transparently reopens
     Failure/Error: second_socket_id = client.get("#{server.endpoint}/socket").body.to_s
     IOError:
       problem making HTTP request: End of file reached
     Shared Example Group: "HTTP handling" called from ./spec/lib/http/client_spec.rb:219
     # ./lib/http/connection.rb:99:in `read_headers!'
     # ./lib/http/client.rb:77:in `make_request'
```

**Comment by tarcieri** [comment]
This was "fixed" in #233, but the root cause is probably worth revisiting.
